src/ripple/server/Port.h

```diff
@@ -97,11 +98,16 @@ inline
 std::ostream&
 operator<< (std::ostream& os, Port const& p)
 {
-    os <<
-        "'" << p.name <<
-        "' (ip=" << p.ip << ":" << p.port <<
-        (p.allow_admin ? ", admin, " : ", ") <<
-        p.protocols() << ")";
+    os << "'" << p.name << "' (ip=" << p.ip << ":" << p.port << ", ";
+
+    if (! p.admin_ip.empty ())
+    {
+        os << "admin IPs:";
+        for (auto const& ip : p.admin_ip)
+            os << ip.to_string () << ", ";
+    }
```

Comment:
It's a shame C++ doesn't have a nice way to stream a comma separated list. I don't object to the extra comma (unless there's a terse way that you can think of).